### PR TITLE
:sparkles: Session sharing functionality was implemented.

### LINKED ISSUE #377: Share button should enabled
**Idea Description**
Currently share button is not working.
Please enable share function.

**Reference images and links**
258797112-9d2f6fca-333b-4cf8-bc35-f50dd0ae5a06

### PR DESCRIPTION
**Pull request by Corvus400**
## Issue
- close #377 

## Overview (Required)
- The sharing functionality implemented at last year's Droidkaigi has been implemented again this year.

## Links
- https://github.com/DroidKaigi/conference-app-2022/pull/239
- https://github.com/DroidKaigi/conference-app-2022/pull/334

## Movie (Optional)
Before | After
:--: | :--:
--- | <video src="https://github.com/DroidKaigi/conference-app-2023/assets/13657682/84de5fa4-0ce2-4aea-8346-3cc9af79ff4c" width="300" >

### CODE REVIEW

File: app-android/src/main/java/io/github/droidkaigi/confsched2023/share/AndroidShareManager.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+class AndroidShareManager(private val context: Context) : ShareManager {
+    override fun share(text: String) {
+        try {
+            ShareCompat.IntentBuilder(context)
+                .setText(text)
+                .setType("text/plain")
+                .startChooser()
+        } catch (e: ActivityNotFoundException) {
+            Log.e("ActivityNotFoundException", "Fail startActivity", e)
```
Comment (takahirom) [inline app-android/src/main/java/io/github/droidkaigi/confsched2023/share/AndroidShareManager.kt:16]: We use Kermit for logging. Please check it out 🙏

File: core/common/src/commonMain/kotlin/io/github/droidkaigi/confsched2023/share/ShareManager.kt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package io.github.droidkaigi.confsched2023.share
+
+/**
+ * Platform sharing feature wrapper
+ */
+interface ShareManager {
```
Comment (takahirom) [inline core/common/src/commonMain/kotlin/io/github/droidkaigi/confsched2023/share/ShareManager.kt:6]: Maybe we won't use this this year for other platforms. So how about removing this and rename AndroidShareManager with `ShareStarter`or `ShareLauncher` or `ShareNavigator`?

File: app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt [hunk 3]
```diff
@@ -284,6 +292,14 @@ private class ExternalNavController(
         context.startActivity(Intent(context, OssLicensesMenuActivity::class.java))
     }
 
+    fun onShareClick(timeTableItem: TimetableItem) {
+        shareNavigator.share(
+            "[${timeTableItem.room.name.currentLangTitle}] ${timeTableItem.startsTimeString} - ${timeTableItem.endsTimeString}\n" +
```
Comment (tkhs0604) [inline app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt:297]: How about adding date info too? 👀
Perhaps, we are referring the formatted date property on the detail screen, so we may reuse it.

**Review by takahirom — APPROVED** [review]
We can ~tweet~ post by this! Thanks! 🐦

**Comment by takahirom** [comment]
@tkhs0604 Please merge when you like!

File: app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt (hunk 3 above)
Comment (Corvus400) [reply app-android/src/main/java/io/github/droidkaigi/confsched2023/KaigiApp.kt]: @tkhs0604 
It seems that we can add a method to get only month and date by adding it to TimetableItem.kt.
However, I felt it was a bit redundant to add a method that is only used for the share function, so I decided to revert to the original method. :bow:

**Review by tkhs0604 — APPROVED** [review]
Sorry for bothering you, and thank you for implementing nice share function! 📪

**Pull request merged**